components/ws-daemon/pkg/content/service.go

```diff
@@ -763,6 +770,40 @@ func workspaceLifecycleHooks(cfg Config, kubernetesNamespace string, workspaceEx
 	return map[session.WorkspaceState][]session.WorkspaceLivecycleHook{
 		session.WorkspaceInitializing: {setupWorkspace, startIWS},
 		session.WorkspaceReady:        {setupWorkspace, startIWS},
-		session.WorkspaceDisposing:    {iws.StopServingWorkspace},
+		//session.WorkspaceDisposing:    {iws.StopServingWorkspace},
+		session.WorkspaceDisposed: {iws.StopServingWorkspace},
 	}
 }
+
+// if the mark mount still exists in /proc/mounts it means we failed to unmount it and
+// we cannot remove the content. As a side effect the pod will stay in Terminating state
+func unmountMark(path string) error {
+	procMounts, err := ioutil.ReadFile("/proc/mounts")
+	if err != nil {
+		log.WithError(err).Error("cannot read /proc/mounts")
+		return err
+	}
+
+	scanner := bufio.NewScanner(bytes.NewReader(procMounts))
```

Comment:
It would be really nice to have a couple of tests for this parser.

Reply:
done

Reply:
we've had this parser in the codebase before, including unit tests. The "ws-daemon sandbox" mechanism looked very similar.

Also, I think the prometheus procfs package already has a parser for this file.